libnmstate/netapplier.py

```diff
@@ -118,13 +118,15 @@ def _apply_ifaces_state(
     desired_state, verify_change, commit, rollback_timeout
 ):
     current_state = state.State(netinfo.show())
-
-    desired_state.sanitize_ethernet(current_state)
-    desired_state.sanitize_dynamic_ip()
-    desired_state.merge_routes(current_state)
-    desired_state.merge_dns(current_state)
-    metadata.generate_ifaces_metadata(desired_state, current_state)
-
+    vlan_ifaces_state = _prepare_vlan_ifaces_for_mtu_change(
```

Comment:
This fits now as a ‘state.State’ method like the others.
‘desired_state.compement_vlan_interfaces(current_state)’
It should identify which current vlan ifaces are not mentioned in the desired state but their base is, and add their name and maybe the state/up (it is enough) to the desired state.
Nothing more should be needed, because later on the flow, a merge will occur anyway.

This new method can now be unit tested easily in the state test module.

Reply:
Wouldn't it be better to directly include all VLANs that are not in the desired state?

Reply:
Doing that will touch unnecessary ifaces.. why do it?

Reply:
Not needed anymore. I've deleted it.